warehouse/admin/views/users.py

```diff
@@ -295,6 +295,41 @@ def user_reset_password(user, request):
     return HTTPSeeOther(request.route_path("admin.user.detail", username=user.username))
 
 
+@view_config(
+    route_name="admin.user.wipe_factors",
+    require_methods=["POST"],
+    permission="admin",
+    has_translations=True,
+    uses_session=True,
+    require_csrf=True,
+    context=User,
+)
+def user_wipe_factors(user, request):
+    if user.username != request.matchdict.get("username", user.username):
+        return HTTPMovedPermanently(request.current_route_path(username=user.username))
+
+    if user.username != request.params.get("username"):
+        request.session.flash("Wrong confirmation input", queue="error")
+        return HTTPSeeOther(
+            request.route_path("admin.user.detail", username=user.username)
+        )
+
+    user.totp_secret = None
+    user.webauthn = []
+    user.recovery_codes = []
+
+    login_service = request.find_service(IUserService, context=None)
+    send_password_compromised_email(request, user)
+    login_service.disable_password(
```

Comment:
I think we want to initiate a password reset flow (including sending the email).

We should be able to just call user_reset_password to accomplish that.

Reply:
Yeah, I just copied this directly from the body of `user_reset_password`, since I wasn't sure about calling a `view_config`-decorated function within another. If that's okay though, I can replace it!

Reply:
0ba6bc02